ChatSX/components/Poll/index.tsx

```diff
@@ -0,0 +1,1689 @@
+      } else {
+        const selectedTime = selectedValue || newDate;
+        setTime(selectedTime);
+        setShow(false);
+        setMode(DATE_TEXT);
+      }
+    }
+  };
+
+  // this function changes the mode
+  const showMode = (currentMode: any) => {
+    setShow(true);
+    setMode(currentMode);
+  };
+
+  // this function set mode "date"
+  const showDatePicker = () => {
+    showMode(DATE_TEXT);
+  };
+
+  // this function handles question input in poll
+  const handleQuestion = (val: string) => {
+    setQuestion(val);
+  };
+
+  // this function toggles the advance button to show and hide advance options
+  const handleShowAdvanceOption = (val: boolean) => {
+    setShowAdvancedOption(!showAdvancedOption);
+  };
+
+  // this function handles "allow voters to add more options" toggle button
+  const handleAddOptions = (val: boolean) => {
+    setAddOptionsEnabled(val);
+  };
+
+  // this function handles "Anonymous poll" toggle button
+  const handleAnonymousPoll = (val: boolean) => {
+    setAnonymousPollEnabled(val);
+  };
+
+  // this function handles "Anonymous poll" toggle button
+  const handleLiveResults = (val: boolean) => {
+    setLiveResultsEnabled(val);
+  };
+
+  // this function handles "User can vote for" modal dropdown selection
+  const handleUserVoteFor = (val: number) => {
+    setUserVoteFor(val);
+    hideActionModal();
+  };
+
+  // this function handles "Number of votes allowed per user" modal dropdown selection
+  const handleVoteAllowedPerUser = (val: number) => {
+    setVoteAllowedPerUser(val + 1);
+    hideSelectOptionModal();
+  };
+
+  // this function handles "Cancel" button flow
+  const handleOnCancel = () => {
+    navigation.goBack();
+  };
+
+  // this function hides "User can vote for" modal
+  const hideActionModal = () => {
+    setIsActionAlertModalVisible(false);
+  };
+
+  // this function hides "Number of votes allowed per user" modal
+  const hideSelectOptionModal = () => {
+    setIsOptionAlertModalVisible(false);
+  };
+
+  // this function sets "User can vote for" modal interated value to the state
+  const handleOpenActionModal = () => {
+    let valueArr: any = userCanVoteForArr;
+
+    setUserVoteForOptionsArrValue(valueArr);
+    setIsActionAlertModalVisible(true);
+  };
+
+  const handleShowDateTimePicker = () => {
+    setShow(false);
+    setMode('');
+    setDate('');
+    setTime(new Date());
+  };
+
+  // this function sets "Number of votes allowed per user" modal interated value to the state
+  const handleOpenOptionModal = () => {
+    let quantinyArr: any = [];
+    for (let i = 0; i < optionsArray.length; i++) {
+      quantinyArr = [
+        ...quantinyArr,
+        `${i + 1 > 1 ? `${i + 1} options` : `${i + 1} option`}`,
+      ];
+    }
+    setUserVoteForOptionsArrValue(quantinyArr);
+    setIsOptionAlertModalVisible(true);
+  };
+
+  // this function fetches postPollConversation API
+  async function postPoll() {
```

Comment:
why we are making API call in UI class, by this way, will not be able to segregate UI and Business logic